src/components/ListItem.jsx

```diff
@@ -1,5 +1,41 @@
+import React, { useState, useEffect } from 'react';
 import './ListItem.css';
+import { updateItem } from '../api/firebase';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listPath }) {
+	const { id, totalPurchases, name, dateLastPurchased } = item;
+	const lessThan24HoursSincePurchase =
+		dateLastPurchased &&
+		(new Date() - dateLastPurchased.toDate()) / 3600000 < 24;
+
+	const [isChecked, setIsChecked] = useState(
+		lessThan24HoursSincePurchase || false,
+	);
+
+	async function handleCheckboxCheck() {
+		setIsChecked(!isChecked);
+		if (!isChecked) {
+			try {
+				await updateItem(listPath, {
```

Comment:
note: something worth considering in React apps. Handling these states in an app in order is important for showing the user what is going on.

- Loading
- Error
- Success with no data
- Success with data

Right now we are only showing the user when data is successful. I know this goes out of the scope of the issue but this is something important to keep in mind for a fully complete app.

This is one reason why things like React query exists to help handle these states much easier:

<img width="559" alt="image" src="https://github.com/the-collab-lab/tcl-67-smart-shopping-list/assets/19897674/f1b11f64-f624-462b-9833-2a8ab3ca8459">

Reply:
We chatted about it briefly in office hours, but error boundaries are also a decent approach to handle component “failures”. The point remains - you want to think about more than just “the happy path”, as noted above

Reply:
Thanks Britt and Aditya, I've attempted to implement the `useQuery` hook and it seems to be working, however it is hard to tell since the request cycle is happening very quickly. 
Also, if I am using `useQuery`, do I still need the try catch, or is that redundant since we are using `useQuery`?

Reply:
I believe you can just check the error object returned from useQuery, however I haven’t personally used this library

Reply:
@jtkabenni I opened up a quick [PR](https://github.com/the-collab-lab/tcl-67-smart-shopping-list/pull/39) to show how to use React query. React query is not a bad library to learn. So as a quick explanation of the `useQuery` and `useMutation` pattern.

`useQuery`: is for `GET` requests

`useMutation`: is for `PUT`, `UPDATE`, `DELETE` (things that mutate)

I included in the solution in the PR I created but in this case you'd want a `useMutation`. It is slightly weird because useQuery uses `isFetching` and useMutation uses `isLoading` for loading states. Also you don't need a try catch when using the call since react query will automatically catch it for you.

Also another way to do this without react query just as reference would be have these useState calls:

```javascript
const [isLoading, setIsLoading] = useState(false);
const [data, setData] = useState(undefined);
const [error, setError] = useState(undefined);

const handleChecked = () => {
       setIsLoading(true);
       await updateItem(listItem, {
          itemId: id,
          totalPurchases
       })
        .catch(e => setError(e))
        .finally(() => setIsLoading(false))
}
```

It's just a lot to type out and is one of the reasons things like React query exist to get rid of the amount you have to type out. But wanted to share how this could be accomplished without React query.

If there are any other questions let me know but it's awesome you've added React query!

Reply:
Thanks Britt, this is really helpful and thanks for showing how it works that other PR too. I've made those updates.
I wonder why useQuery worked when I used it instead of useMutation